gateway-export-plugin/src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/writer/PolicyWriter.java

```diff
@@ -49,28 +51,95 @@ public void write(Bundle bundle, File rootFolder) {
 
         Stream.of(
                 bundle.getEntities(Policy.class).values().stream(),
                 bundle.getEntities(GlobalPolicy.class).values().stream().map(Policy.class::cast).collect(toList()).stream(),
                 bundle.getEntities(AuditPolicy.class).values().stream().map(Policy.class::cast).collect(toList()).stream()
         ).flatMap(s -> s)
-                .forEach(policyEntity -> writePolicy(bundle, policyFolder, policyEntity.getParentFolder().getId(), policyEntity.getName(), policyEntity.getPolicyDocument()));
+                .forEach(policyEntity -> {
+                    final String id = policyEntity.getId();
+                    final String name = policyEntity.getName();
+                    final PolicyMetadata policyMetadata = writePolicy(bundle, policyFolder, policyEntity.getParentFolder().getId(), name, id, policyEntity.getPolicyDocument());
+                    final Set<Dependency> policyDependencies = getPolicyDependencies(id, rawBundle);
+                    policyMetadata.setUsedEntities(policyDependencies);
+                    final PolicyType policyType = policyEntity.getPolicyType();
+                    if(policyType != null){
+                        policyMetadata.setType(policyType.getType());
+                    }
+                    policyMetadata.setTag(policyEntity.getTag());
+                    policyMetadata.setSubtag(policyEntity.getSubtag());
+                    policyMetadataMap.put(policyMetadata.getNameWithPath(), policyMetadata);
+                });
+        writePolicyMetadata(policyMetadataMap, policyFolder);
+    }
+
+    /**
+     * Writes policy metadata to policy.yml file
+     * @param policyMetadataMap
+     * @param policyFolder
+     */
+    private void writePolicyMetadata(final Map<String, PolicyMetadata> policyMetadataMap, final File policyFolder) {
+        if (!policyMetadataMap.isEmpty()) {
+            //build yml file with dependencies
+            File policyMetadataFile = new File(policyFolder, "policies" + jsonTools.getFileExtension());
+            try {
+                jsonTools.writeObject(policyMetadataMap, policyMetadataFile);
+            } catch (IOException e) {
+                throw new WriteException("Error writing policy metadata file", e);
+            }
+        }
     }
 
-    private void writePolicy(Bundle bundle, File policyFolder, String folderId, String name, Element policy) {
+    private Set<Dependency> getPolicyDependencies(final String id, final Bundle rawBundle){
+        final Set<Dependency> dependencyList = new HashSet<>();
+        Map<Dependency, List<Dependency>> dependencyListMap = rawBundle.getDependencyMap();
+        if (dependencyListMap != null) {
+            populateDependencies(dependencyListMap, id, id, dependencyList);
+        }
+        return dependencyList;
+    }
+
+    private void populateDependencies(Map<Dependency, List<Dependency>> dependencyListMap, String rootId, String id, Set<Dependency> dependencies) {
```

Comment:
How is this different from BundleBuilder.buildDependencies? 
Wouldn't it be possible to use it? So that, we could avoid passing [raw-bundle].

Reply:
BundleBuilder parses raw xml bundle and populates Bundle.java with all the dependencies
so we dont have to parse the xml again instead we can use the dependencies from the Bundle java object itself

Reply:
Got it.